packages/utils/src/memo.ts

```diff
@@ -40,5 +40,5 @@ export function memoBuilder(): MemoFunc {
       }
     }
   }
-  return [memoize, unmemoize];
+  return { memoize, unmemoize };
```

Comment:
The tuple return was done for bundle size reasons btw - I think it's reasonable enough to leave as is as long as we comment it properly.

Reply:
I know, but IMHO we've gone too far with bundle size when we let it make our code confusing to read, which it was when I first was looking into this.

Reply:
Thats a fair point. Can we perhaps use array destructuring to avoid the `memo[0]` calls?

```ts
const [memoize, unmemoize] = memo;
```

Reply:
Sure, here I think that's fine. I'll make the change. In general, though, I'm of the mind that bundle size is important but that I also don't want to get to a point where we feel like we can't use objects when they make sense.

Help me remember, though, please. As I understand it, the reason a destructured array is better than an object (or even a destructured object) is that terser won't mangle the property names because they count as public API, whereas with an array the local names used in the destructuring will get mangled? Would it still count as public API if `memoBuilder` were to be defined in the same file where it's used (and were not to be exported, since it's not used anywhere else)? No, right?

(I'm asking partly just so I understand what we're talking about. I recognize that making that change would also remove `memoBuilder` from `@sentry/utils`'s public API, and so would have to wait for v7 if it were to happen at all. But if I'm right, it might be worth looking at what else we export from utils which we don't actually need to.)

Reply:
Regardless of if `memoBuilder` is exported or not, `{ memoize, unmemoize }` means that `memoize` and `unmemoize` are unminifiable. The array destructuring just helps with that.